pyscf/solvent/hessian/pcm.py

```diff
@@ -0,0 +1,233 @@
+    grid_coords  = pcmobj.surface['grid_coords']
+    exponents    = pcmobj.surface['charge_exp']
+
+    atom_coords = mol.atom_coords(unit='B')
+    atom_charges = numpy.asarray(mol.atom_charges(), dtype=numpy.float64)
+    fakemol_nuc = gto.fakemol_for_charges(atom_coords)
+    fakemol = gto.fakemol_for_charges(grid_coords, expnt=exponents**2)
+
+    # nuclei potential response
+    int2c2e_ip1ip2 = mol._add_suffix('int2c2e_ip1ip2')
+    v_ng_ip1ip2 = gto.mole.intor_cross(int2c2e_ip1ip2, fakemol_nuc, fakemol).reshape([3,3,mol.natm,-1])
+    dv_g = numpy.einsum('n,xyng->ngxy', atom_charges, v_ng_ip1ip2)
+    dv_g = numpy.einsum('ngxy,g->ngxy', dv_g, q_sym)
+
+    de = numpy.zeros([mol.natm, mol.natm, 3, 3])
+    for ia in range(mol.natm):
+        p0, p1 = gridslice[ia]
+        de_tmp = numpy.sum(dv_g[:,p0:p1], axis=1)
+        de[:,ia] -= de_tmp
+        #de[ia,:] -= de_tmp.transpose([0,2,1])
+
+
+    int2c2e_ip1ip2 = mol._add_suffix('int2c2e_ip1ip2')
+    v_ng_ip1ip2 = gto.mole.intor_cross(int2c2e_ip1ip2, fakemol, fakemol_nuc).reshape([3,3,-1,mol.natm])
+    dv_g = numpy.einsum('n,xygn->gnxy', atom_charges, v_ng_ip1ip2)
+    dv_g = numpy.einsum('gnxy,g->gnxy', dv_g, q_sym)
+
+    for ia in range(mol.natm):
+        p0, p1 = gridslice[ia]
+        de_tmp = numpy.sum(dv_g[p0:p1], axis=0)
+        de[ia,:] -= de_tmp
+        #de[ia,:] -= de_tmp.transpose([0,2,1])
+
+    int2c2e_ipip1 = mol._add_suffix('int2c2e_ipip1')
+    v_ng_ipip1 = gto.mole.intor_cross(int2c2e_ipip1, fakemol_nuc, fakemol).reshape([3,3,mol.natm,-1])
+    dv_g = numpy.einsum('g,xyng->nxy', q_sym, v_ng_ipip1)
+    for ia in range(mol.natm):
+        de[ia,ia] -= dv_g[ia] * atom_charges[ia]
+
+    v_ng_ipip1 = gto.mole.intor_cross(int2c2e_ipip1, fakemol, fakemol_nuc).reshape([3,3,-1,mol.natm])
+    dv_g = numpy.einsum('n,xygn->gxy', atom_charges, v_ng_ipip1)
+    dv_g = numpy.einsum('g,gxy->gxy', q_sym, dv_g)
+    for ia in range(mol.natm):
+        p0, p1 = gridslice[ia]
+        de[ia,ia] -= numpy.sum(dv_g[p0:p1], axis=0)
+
+    return de
+
+def hess_elec(pcmobj, dm, verbose=None):
+    '''
+    slow version with finite difference
+    TODO: use analytical hess_nuc
+    '''
+    log = logger.new_logger(pcmobj, verbose)
+    t1 = (logger.process_clock(), logger.perf_counter())
+    pmol = pcmobj.mol.copy()
+    mol = pmol.copy()
+    coords = mol.atom_coords(unit='Bohr')
+
+    def pcm_grad_scanner(mol):
+        # TODO: use more analytical forms
+        pcmobj.reset(mol)
+        e, v = pcmobj._get_vind(dm)
+        pcm_grad = grad_nuc(pcmobj, dm)
+        pcm_grad+= grad_solver(pcmobj, dm)
+        pcm_grad+= grad_qv(pcmobj, dm)
+        return pcm_grad
+
+    mol.verbose = 0
+    de = numpy.zeros([mol.natm, mol.natm, 3, 3])
+    eps = 1e-3
+    for ia in range(mol.natm):
+        for ix in range(3):
+            dv = numpy.zeros_like(coords)
+            dv[ia,ix] = eps
+            mol.set_geom_(coords + dv, unit='Bohr')
+            mol.build()
```

Comment:
`mol.build()` is not needed for set_geom_ function.

Reply:
removed